qiskit_quantuminspire/api/authentication.py

```diff
@@ -0,0 +1,62 @@
+import time
+from typing import Any, Tuple, cast
+
+import requests
+
+from qiskit_quantuminspire.api.settings import ApiSettings, TokenInfo, Url
+
+
+class AuthorisationError(Exception):
+    """Indicates that the authorisation permanently went wrong."""
+
+    pass
+
+
+class IdentityProvider:
+    def __init__(self, well_known_endpoint: str):
+        self._well_known_endpoint = well_known_endpoint
+        self._token_endpoint, self._device_endpoint = self._get_endpoints()
+        self._headers = {"Content-Type": "application/x-www-form-urlencoded"}
+
+    def _get_endpoints(self) -> Tuple[str, str]:
```

Comment:
Shall we use `@cache` or it might be overkill ?

Reply:
Only called once for initialization, so currently there is not really a point in (also) caching using the functools decorator.